lib/ansible/modules/utilities/logic/async_wrapper.py

```diff
@@ -232,14 +233,13 @@ def _run_module(wrapped_cmd, jid, job_path):
     jobdir = os.path.expanduser(async_dir)
     job_path = os.path.join(jobdir, jid)
 
-    if not os.path.exists(jobdir):
-        try:
-            os.makedirs(jobdir)
-        except Exception:
-            print(json.dumps({
-                "failed": 1,
-                "msg": "could not create: %s" % jobdir
-            }))
+    try:
+        makedirs_safe(jobdir)
```

Comment:
fyi, this does not 'fix' race condition, it just makes it both less probable and ignore the error if it already exists

Reply:
It's not an error if it already exists, because I _know_ that it already exists, and I know why. It's because it's ran locally against >1 host, so multiple processes are creating the same parent directory, but will create a randomized subdir. There is no issue with this, and it works just fine when the simultaneous processes do not hit the race condition.